src/ServiceControl/MessageRedirects/Api/MessageRedirectsModule.cs

```diff
@@ -65,7 +67,7 @@ public MessageRedirectsModule()
                     await collection.Save(session).ConfigureAwait(false);
                 }
 
-                Bus.Publish(new MessageRedirectCreated
+                DomainEvents.Raise(new MessageRedirectCreated
```

Comment:
What are the potential consequences of this for the callers of the API? Previously this was a publish operation which means the actual handling of the message is decoupled from API call. Now, these two things are coupled together which means a potential failure of handling that message could bubble up to the caller of the API.

Reply:
True. Do you think we should do something about this on SP level? i.e. retry if API call failed or something?

Reply:
Good question. I don't know SC enough to make a judgment call on this one. Maybe @WilliamBZA or @boblangley can join into the discussion

Reply:
@danielmarbach is this the last thing we need to decide on?

Reply:
I think we do

Reply:
@WilliamBZA @boblangley @tmasternak what do you think?

Reply:
Hmmm. This always had the potential for transactional inconsistency. In other places in the code we accept the user request and turn it into a message so that the handling of the message will go through FLR. Does Pulse use the reason phrases and distinguish all of the error conditions that we detect here?

Reply:
Having said that, the document session is not transactionally consistent with the transport anyway. 

In this case, if there's an error in the Domain Event handlers, then that error will get bubbled all the way to Nancy where it will get turned into a 404. If we successfully created the redirect then a retry will just return a "Duplicate". The rest of the Domain Event handlers and the SendLocal will not get sent. 

How bad is that (in this situation)?

Reply:
Pulse does not. In fact there is almost zero error handling in pulse besides in the monitoring code.

The transport that the publish uses to send messages to pulse (SignalR) is also not safe, and these publish events are at most-once delivery anyway.

Reply:
TL;DR: from the pulse side there is no transactional guarantee around the publish portion anyway. Even if we had full messaging in place at the server side with FLR+SLR, we'd still have to build outbox into the signlar hubs to ensure we don't miss messages during reconnect/negotiation/connection upgrade/etc.

Reply:
I think that Daniel's original concern was more around the fact that there is more stuff happening now as part of request handling and the likelihood of Pulse showing an error page is higher.

Reply:
I'm good with that. It was reasonably discussed. Let's move on

Reply:
It won't be any more buggy than it currently is, but the value is higher. So 👍 merge it

Reply:
Oops, it already is...